### ISSUE TITLE
Gateway failed to get OAuth2 tokens from UAA

### ISSUE THREAD

**Issue by StayHungryStayFoolish** [jhipster/generator-jhipster#12149]
**Overview of the issue**

I used JHipster version 6.10.1 to generate uaa and generated gateway based on uaa. When I start a project cloned based on https://github.com/jhipster/jhipster-registry or a Docker image based on jhipster/jhipster-registry latest, Start uaa again, when the gateway starts, the following error is displayed. Regitry displays Uaa and Gateway instances, but it seems that Gateway cannot be accessed with Uaa through instanceId. Because Gateway cannot be accessed through **http://uaa/oauth/token_key**, but you can use **http://localhost:9999/oauth/token_key** to get the **PUBLIC KEY**.

**Reproduce the error**
```java
2020-07-26 14:30:13.868 ERROR 2157 --- [  restartedMain] c.z.b.c.o.OAuth2JwtAccessTokenConverter  : could not get public key from OAuth2 server to create SignatureVerifier

org.springframework.web.client.HttpClientErrorException$NotFound: 404 Not Found: [404 page not found]
	at org.springframework.web.client.HttpClientErrorException.create(HttpClientErrorException.java:113)
	at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:170)
	at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:112)
	at org.springframework.web.client.ResponseErrorHandler.handleError(ResponseErrorHandler.java:63)
	at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:782)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:740)
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:674)
	at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:583)
	at com.sample.test.security.oauth2.UaaSignatureVerifierClient.getSignatureVerifier(UaaSignatureVerifierClient.java:46)
	at com.sample.test.config.oauth2.OAuth2JwtAccessTokenConverter.tryCreateSignatureVerifier(OAuth2JwtAccessTokenConverter.java:69)
	at com.sample.test.config.oauth2.OAuth2JwtAccessTokenConverter.<init>(OAuth2JwtAccessTokenConverter.java:29)
	at com.sample.test.config.SecurityConfiguration.jwtAccessTokenConverter(SecurityConfiguration.java:68)
	at com.sample.test.config.SecurityConfiguration$$EnhancerBySpringCGLIB$$ff68d821.CGLIB$jwtAccessTokenConverter$2(<generated>)
	at com.sample.test.config.SecurityConfiguration$$EnhancerBySpringCGLIB$$ff68d821$$FastClassBySpringCGLIB$$e3fca0d4.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:331)
	at com.sample.test.config.SecurityConfiguration$$EnhancerBySpringCGLIB$$ff68d821.jwtAccessTokenConverter(<generated>)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:651)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:636)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1338)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1177)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1306)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1226)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:885)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:789)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:539)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1338)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1177)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1306)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1226)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:885)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:789)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:228)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1358)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:895)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:878)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:550)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:747)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
	at com.sample.test.GatewayApp.main(GatewayApp.java:65)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49)
```

**JHipster Version(s)**

**jhipster --version**
INFO! Using JHipster version installed globally
6.10.1
**JHipster Regitry**
- **Docker Image: latest**
- **Source Code:https://github.com/jhipster/jhipster-registry**

**JHipster configuration**

**Eureka version : spring-cloud-starter-netflix-eureka-client : 2.2.7.RELEASE**

`Gateway yo-rc.json`

```json
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.sample.test",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.10.1",
    "applicationType": "gateway",
    "baseName": "gateway",
    "packageName": "com.sample.test",
    "packageFolder": "com/sample/test",
    "serverPort": "8080",
    "authenticationType": "uaa",
    "uaaBaseName": "uaa",
    "cacheProvider": "ehcache",
    "enableHibernateCache": false,
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": "eureka",
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "embeddableLaunchScript": false,
    "useSass": true,
    "clientPackageManager": "npm",
    "clientFramework": "react",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "creationTimestamp": 1595742672959,
    "testFrameworks": ["gatling"],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en"],
    "blueprints": []
  }
}
```

`Uaa yo-rc.json`

```json
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.sample.test",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.10.1",
    "applicationType": "uaa",
    "baseName": "uaa",
    "packageName": "com.sample.test",
    "packageFolder": "com/sample/test",
    "serverPort": "9999",
    "authenticationType": "uaa",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": "eureka",
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "YmMwMzFhY2UyNGQ2YTJmNGIwYjY0ZjhlY2QyNjhhNGZkMWZjZDhjNjYwNGUxMmU2MzdlNzAxMWE5MmJmNmIxMDY3M2M5ZTdkOGVhZGNiOGNmNzQwNzI0OGNlNDNkYmUyZTY0MTdiYWM5NDM0Y2RjMjhmYzVjYWQ4NjNjMTNlOTc=",
    "embeddableLaunchScript": false,
    "creationTimestamp": 1595740840514,
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": true,
    "clientPackageManager": "npm",
    "nativeLanguage": "en",
    "languages": ["en"],
    "blueprints": [],
    "skipClient": true
  }
}
```

**Comment by StayHungryStayFoolish** [comment]
I rolled back the JHipster version to 6.5.1 and everything is fine.

**Comment by chelfe** [comment]
Hello, I can't reproduce your issue with your uaa and gateway yo-rc.json.
I run it with the same **jHipster version (6.10.1)** and **latest jHipster registry**.

I don't have any error and I can log in to gateway with admin account for example.

Can you regenerate uaa and gateway, and give us more details to help us to reproduce your issue ?

**Comment by pascalgrimaud** [comment]
as we can't reproduce, I think it's fixed in v6.10.1
let's close this

Don't hesitate to give a try and tell us if you can reproduce in last version
